src/lib.rs

```diff
@@ -237,9 +246,9 @@ pub fn derive(input: TokenStream) -> TokenStream {
 
     let ast = syn::parse_macro_input(&input).expect("Couldn't parse item");
 
-    let result = builder_for_struct(ast);
+    let result = builder_for_struct(ast).to_string();
 
-    format!("{}", result).parse().expect("Couldn't parse string to tokens")
+    result.parse().expect(&format!("Couldn't parse {:?} to tokens", result))
```

Comment:
done - I switched from {:?} to `{}` there.

I also changed other places from {:?} to `{:?}`, because most syn-Items only have the fmt::Debug trait.

Until now I prefered {:?} over `{}`, because it is less ambiguous. It's either
```
Couldn't parse "struct \"foo\" {}" to tokens: LexError { _inner: () }
```

or
```
Couldn't parse `struct "foo" {}` to tokens: LexError { _inner: () }
```

The second variant looks nicer - but becomes ambiguous if the token stream itself contains a back tick. => I guess it's a trade off worthwhile.